### PR TITLE
Fixes delph-in#671 - bugfix for multiple inheritance for lexical types

### PR DESCRIPTION
**Pull request by Ubadub**
See discussion in #671 

@ltxom let me know if anything is not satisfactory or if the regression tests run into issues. Thank you!

### CODE REVIEW

File: gmcs/linglib/lexicon.py [hunk 1]
```diff
@@ -1080,22 +1097,16 @@ def validate_lexicon(ch, vr):
...
-                                       + lextype + ' class. If the value specified here is conflict with that other' \
-                                              ' specification, the grammar will not compile.'
-                                vr.warn(feat.full_key+'_name', mess)
+    # Abhinav 2023-01-31: Fixed bug blocking multiple inheritance, no longer recursive
+    # so no need to check for cycles (see discussion in issue #671)
+    for lextype in ALL_LEX_TYPES:
+        for lt in ch.get(lextype):
+            feat_name2full_key = {feat.get('name') : feat.full_key for feat in lt.get('feat') if feat}
+            supertype_feat_names = { feat.get('name', '') for feat in get_all_supertypes_features(ch, lt) }
+            dupe_feats = supertype_feat_names.intersection(feat_name2full_key.keys())
```
Comment (dantiston) [inline gmcs/linglib/lexicon.py]:
```suggestion
            feat_name2full_key = {feat.get('name'): feat.full_key for feat in lt.get('feat') if feat}
            supertype_feat_names = {feat.get('name', '') for feat in get_all_supertypes_features(ch, lt)}
            dupe_feats = supertype_feat_names.intersection(feat_name2full_key)
```

1. `dict`s iterate through their keys, so `supertype_feat_names.intersection(feat_name2full_key.keys())` is redundant to `supertype_feat_names.intersection(feat_name2full_key)`.
2. Styling: there shouldn't be a space before the colon on line 1104 or before or after the curly braces on 1105.

Comment (Ubadub) [reply gmcs/linglib/lexicon.py]: Thanks! Yes, I had realized that about the dictionary keys and meant to change it, but forgot to before committing/PRing this. Good catch!

**Comment by ltxom** [comment]
Thank you. I verified these changes, which don't break other parts of the existing regression tests, and it solved the errors when validating  [choices_supertype_bugreport.txt](https://github.com/delph-in/matrix/files/10541561/choices_supertype_bugreport.txt).

@emilymbender Since it is a fix relating to the validation process, should we introduce a new regression test for it?

**Comment by ltxom** [comment]
Emily mentioned a unit test could be added to verify validation improvements. The section about how to run/add a unit test is [here](https://github.com/delph-in/docs/wiki/MatrixCustomizationUnitTesting).

However, I followed the documentation above and found the parameter of "unit-test" is missing in the current Matrix trunk. I checked the commit history and found [Dr. Goodman updated it](https://github.com/delph-in/matrix/commit/dcb679a58569bd213e4e15d5a3b1dea8a3dbc9fe) to pytest.

> Tests can be run with `pytest`. For now this requires setting the
> PYTHONPATH variable:
> 
>     PYTHONPATH=. pytest tests/unit/choices_test.py
>     PYTHONPATH=. pytest tests/unit/validate_test.py
> 
> Don't just run all tests under `tests/unit/` because some are still
> broken. The goal is to rework these tests under a more standard pytest
> workflow so simply running `pytest` will be enough.

@Ubadub Do you have experience using pytest and adding a unit test?

**Comment by Ubadub** [comment]
I tried running unit tests a couple days ago but ran into precisely the issue you have clarified above, so thank you for that. Yes, I've used pytest before though not in a few years but I'm sure I can figure it out. However, due to the demands of school it might be a few days before I can get around to it. I will keep you updated!

**Comment by Ubadub** [comment]
Also, as I mentioned in the Issue thread- there actually are regression tests for this, but for customization only, it seems nothing in the regression tests workflow tests the validation script (except indirectly). So maybe we want to change that too? I will write the unit test as soon as I can but if you want me to look into adding a few lines for having regression testing check validate.py as well, I can do that.

**Comment by emilymbender** [comment]
I'm confused as to the goal of this fix. It should be possible to do multiple inheritance on that page, I thought.

**Comment by Ubadub** [comment]
It isn't possible, specifically for lexical types. I've attached a choices file with issue #671 that will allow you to reproduce this issue.

**Pull request merged**